### ISSUE TITLE
Rule proposal: no-duplicate-named

### ISSUE THREAD

**Issue by jfmengels** [benmosher/eslint-plugin-import#598]
I just noticed the the following is considered to be valid syntax by both espree and babel-eslint.

```js
import {a, a, b as a, c as a} from 'foo';
```

We should add a rule that forbids:
- Importing a named export twice: `{a, a}`
- Importing a named export and naming it after another imported named export: `{a, b as a}`
- Importing a named export and naming it after renamed imported named export: `{b as a, c as a}`

To discuss:
- Importing a named export and naming it the original name of an other renamed imported named export: `{b as a, c as b}`

I'm pretty sure we had a rule to forbid `import {b as a} from 'foo'` when `a` is also a named export of that module, but I don't remember which one. If we don't, let me know, and I'll create an issue, because that sounds like a nice style issue to have. That might fill the role of the last `to discuss` item?

I propose the name `no-duplicate-named`, but I'm open to better names :)

**Comment by ntdb** [comment]
I might take a crack at this.

`no-named-as-default` seems to be the closest existing rule to the one you're describing. It prevents the use of a named export's name when importing the default export. [Docs](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-named-as-default.md). Perhaps that rule could be generalized for the case you're describing?

**Comment by jfmengels** [comment]
That'd be great @ntdb! Just wait on a :+1: from @benmosher maybe?

Yes, `no-named-as-default` is very close. I think we can copy most of the logic inside it, but it should be a separate rule, as it would be an unwarranted breaking change IMO (an the name would be misleading).

**Comment by ljharb** [comment]
@ntdb any chance you'd be able to still make this PR?
